### DISCUSSION TITLE
FileNotFoundError

### DISCUSSION THREAD

**Discussion by JK9057** [biomass-dev/biomass#91] [Q&A] [answered]
I am a beginner of Python.

I tried test command.

from biomass import Model, optimize
from biomass.models import Nakakuki_Cell_2010

model = Model(Nakakuki_Cell_2010.__package__).create()

optimize(model, x_id=range(1, 11))

but, it didn't work

error occerd.

FileNotFoundError: [Errno 2] No such file or directory: 'biomass\\models\\Nakakuki_Cell_2010\\set_model.py'

I couldn't find the solution.

Could you give me some advise?

Thank you.

**Comment by himoto** [comment, marked as answer]
Hello @JK9057, thank you for your inquiry, and great thanks for using BioMASS!

I guess you installed BioMASS via pip. Can you please install biomass via **git clone**?

```bash
$ git clone https://github.com/biomass-dev/biomass
$ cd biomass
```

Or download zip and work in `biomass-master` folder.

Then, you can run:

```python
from biomass import Model, optimize
from biomass.models import Nakakuki_Cell_2010

model = Model(Nakakuki_Cell_2010.__package__).create()

optimize(model, x_id=range(1, 11))
```

I hope this helps resolve the issue. If the problem still persists, please do let me know.

Thanks,
Hiroaki

**Reply by JK9057** [reply]
Thank you for your answer.

Yes, I installed via pip.
I could installed via git clone

I have another question.

I changed cd biomass and I entered command that you wrote

Another error is occuerd.
I am so sorry that I could not understand what that was meant

The error message was, (xxxxx is my name)

multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "C:\Users\xxxxx\AppData\Local\Programs\Python\Python39\lib\multiprocessing\pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\xxxxx\biomass\biomass\estimation\ga.py", line 76, in run
    raise FileExistsError(
FileExistsError: Set options['overwrite']=True to overwrite biomass\models\Nakakuki_Cell_2010\out\2
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\xxxxx\biomass\biomass\core.py", line 296, in optimize
    for _ in p.imap_unordered(ga_init.run, x_id):
  File "C:\Users\xxxxx\AppData\Local\Programs\Python\Python39\lib\multiprocessing\pool.py", line 870, in next
    raise value
FileExistsError: Set options['overwrite']=True to overwrite biomass\models\Nakakuki_Cell_2010\out\2

What should I try to solve this problem?

Thank you.

**Reply by himoto** [reply]
Hi @JK9057,

I would like to explain how parameter estimation works in BioMASS.
When you run `optimize()` for the first time, the optimization results will be saved in `out\` folder in the model folder (in your case, it is "C:\Users\xxxxx\biomass\biomass\models\Nakakuki_Cell_2010\out\).

As you may notice, it usually takes a few hours or even a few days to estimate model parameters. To prevent accidental deletion of optimization results, biomass requires users to set options['overwrite'] to True once you get results.

> FileExistsError: Set options['overwrite']=True to overwrite biomass\models\Nakakuki_Cell_2010\out\2

To resolve this error, please run `optimize()` with setting options: overwrite to True (see the following code) or manually remove `out\` in the model folder.

```python
optimize(model, x_id=range(1, 11), options={"overwrite": True})
```

Regards,
Hiroaki

**Reply by JK9057** [reply]
Dear Hiroaki
Thank you for your advice
I really appericate your comment.

It is needed an options (overwrite : True) ! it is very simple but I couldn't imagine.

I retry it.

My PC is not a good performance. It maybe takes a few days.

I will check  that it is finished without any problems.

I am looking forward to being able to use this program.

Best regards, JK.

**Reply by himoto** [reply]
@JK9057,

I recommend you to use the following options:

```python
optimize(
    model, x_id=range(1, 11), options={
        "popsize": 3,
        "max_generation": 100,
        "allowable_error": 0.5,
        "local_search_method": "DE",
        "maxiter": 50,
        "workers": 1,
        "overwrite": True,
    }
)
```
The computation time will also depend on these parameters, especially `max_generation` and `allowable_error`.
Please refer to https://biomass-core.readthedocs.io/en/latest/api/core.html for the description of each parameter.

Please do let me know anytime if you have any questions, I'd be delighted to try to answer questions about BioMASS!

**Reply by JK9057** [reply]
Hello Hiroaki

I appreciate for your comment.
I tried your advice to test, but it took many hours to optimze.
And finally I couldn't finish it.

I think my PC spec is weak, so I changed the parameter.

from biomass import optimize

optimize(
    model, x_id=1, options={
        "popsize": 3,
        "max_generation": 10,
        "allowable_error": 0.5,
        "local_search_method": "DE",
        "maxiter": 50,
        "overwrite": True}
)

Today I found some error messages,
I understood message meanings but I couldn't solve it.

So I hope that you give me some advice.

The message is

C:\Users\XXXX\biomass\biomass\models\Nakakuki_Cell_2010\set_model.py:136: RuntimeWarning: invalid value encountered in double_scalars
  x[C.V57] * y[V.pcFOSn] ** x[C.n57] / (x[C.Km57] ** x[C.n57] + y[V.pcFOSn] ** x[C.n57])
C:\Users\XXXX\biomass\biomass\models\Nakakuki_Cell_2010\set_model.py:107: RuntimeWarning: invalid value encountered in double_scalars
  + (y[V.Fn] / x[C.KF31]) ** x[C.nF31]

py:107: is repeated some time.

Best regards, JK.

**Reply by himoto** [reply]
Hi @JK9057, thank you for your report.
> C:\Users\XXXX\biomass\biomass\models\Nakakuki_Cell_2010\set_model.py:136: RuntimeWarning: invalid value encountered in double_scalars
x[C.V57] * y[V.pcFOSn] ** x[C.n57] / (x[C.Km57] ** x[C.n57] + y[V.pcFOSn] ** x[C.n57])
C:\Users\XXXX\biomass\biomass\models\Nakakuki_Cell_2010\set_model.py:107: RuntimeWarning: invalid value encountered in double_scalars

While estimating parameters, biomass generates multiple parameter candidates. Occasionally they are not suitable for the model to numerically solve ODE. For me, it is reasonable to see this warning during parameter estimation. Since such parameter sets are weeded out during optimization, you do not have to care about this warning right now.

**Reply by JK9057** [reply]
Hello Hioaki,

Thank you for your answer.
I understood that these warnings are not needed to mention.

I am just waiting for finish the optimizing, ain't I?

I would like to know whether BioMass is powerful tool as soon as possible.
